clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy

```diff
@@ -63,6 +64,61 @@ class KubernetesContainerDescription {
   List<KubernetesContainerPort> ports
   KubernetesProbe livenessProbe
   KubernetesProbe readinessProbe
+  List<KubernetesVolumeMount> volumeMounts
+  List<KubernetesEnvVar> envVars
+}
+
+@AutoClone
+@Canonical
+class KubernetesEnvVar {
+  String name
+  String value
+  // TODO(lwander) Q2 2016 add EnvVarSource for selecting secrets.
+}
+
+@AutoClone
+@Canonical
+class KubernetesVolumeMount {
+  String name
+  Boolean readOnly
+  String mountPath
+}
+
+enum KubernetesVolumeSourceType {
+  HOSTPATH, EMPTYDIR, PERSISTENTVOLUMECLAIM, UNSUPPORTED
+}
+
+enum KubernetesStorageMediumType {
+  DEFAULT, MEMORY
+}
+
+@AutoClone
+@Canonical
+class KubernetesVolumeSource {
+  String name
+  KubernetesVolumeSourceType type
+  KubernetesHostPath hostPath
+  KubernetesEmptyDir emptyDir
+  KubernetesPersistentVolumeClaim persistentVolumeClaim
+}
+
+@AutoClone
+@Canonical
+class KubernetesHostPath {
```

Comment:
In the future there will be more load balancing/routing options in these classes, so I'm trying to mirror the kubernetes API structure as closely as possible (as they won't be able to make backwards-incompatible changes) so I don't have to refactor this when the features are added.